src/derivative_utils.jl

```diff
@@ -294,14 +294,14 @@ SciMLBase.isinplace(::WOperator{IIP}, i) where {IIP} = IIP
 Base.eltype(W::WOperator) = eltype(W.J)
 
 set_gamma!(W::WOperator, gamma) = (W.gamma = gamma; W)
-function DiffEqBase.update_coefficients!(W::WOperator, u, p, t)
+function SciMLOperators.update_coefficients!(W::WOperator, u, p, t)
     update_coefficients!(W.J, u, p, t)
     update_coefficients!(W.mass_matrix, u, p, t)
-    W.jacvec !== nothing && update_coefficients!(W.jacvec, u, p, t)
+    !isnothing(W.jacvec) && update_coefficients!(W.jacvec, u, p, t)
     W
 end
 
-function DiffEqBase.update_coefficients!(J::SparseDiffTools.JacVec, u, p, t)
+function SciMLOperators.update_coefficients!(J::FunctionOperator{UJacobianWrapper}, u, p, t)
```

Comment:
@vpuri3 is this right? I thought we could overload `update_coefficients!` on `UJacobianWrapper` directly and it will recursively prop?

Reply:
i think it is working. but the latter might be a neater solution

Reply:
Yeah I think the latter was what I was planning around and should be just a couple lines to implement?

It's also better because this approach is technically type piracy and could lead to invalidations I think (not allowed to own only a type parameter, need to own the type)

Reply:
lets do that later. there are enough complications in this PR anyway.

Reply:
made a note of it in SciMLOps issues